felix/bpf-gpl/tc.c

```diff
@@ -1042,8 +740,30 @@ static CALI_BPF_INLINE struct fwd calico_tc_skb_accepted(struct cali_tc_ctx *ctx
 					ip_hdr(ctx)->protocol == IPPROTO_UDP ? BPF_F_MARK_MANGLED_0 : 0);
 		}
 
-		res |= bpf_l3_csum_replace(skb, l3_csum_off, state->ip_src, state->ct_result.nat_sip, 4);
-		res |= bpf_l3_csum_replace(skb, l3_csum_off, state->ip_dst, state->post_nat_ip_dst, 4);
+		if (!in_place) {
```

Comment:
Any value in wrapping this up as a utility function, pops up in a few places (but maybe calling a utility function would be almost as verbose...

Reply:
yeah, thought about it a long time ago, but not worth it.